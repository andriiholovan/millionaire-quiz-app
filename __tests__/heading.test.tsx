import {
  afterEach,
  describe,
  expect,
  it,
} from 'vitest';
import {
  cleanup,
  render,
} from '@testing-library/react';
import Heading from '../app/_components/heading';

describe('HEADING COMPONENT', () => {
  afterEach(cleanup);

  it('Should match snapshot of component with h1 tag', () => {
    const { asFragment: asFragmentWithDefaultProps } = render(
      <Heading>Who wants to be a millionaire?</Heading>,
    );
    expect(asFragmentWithDefaultProps()).toMatchSnapshot();

    const { asFragment } = render(
      <Heading as="h1">Who wants to be a millionaire?</Heading>,
    );
    expect(asFragment()).toEqual(asFragmentWithDefaultProps());
  });

  it('Should match snapshot of component with h2 tag', () => {
    const { asFragment } = render(
      <Heading
        as="h2"
        className="heading className here"
      >
        Who wants to be a millionaire?
      </Heading>,
    );
    expect(asFragment()).toMatchSnapshot();
  });
});
