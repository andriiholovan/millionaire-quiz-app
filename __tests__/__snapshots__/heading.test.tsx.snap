// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`HEADING COMPONENT > Should match snapshot of component with h1 tag 1`] = `
<DocumentFragment>
  <h1>
    Who wants to be a millionaire?
  </h1>
</DocumentFragment>
`;

exports[`HEADING COMPONENT > Should match snapshot of component with h2 tag 1`] = `
<DocumentFragment>
  <h2
    class="heading className here"
  >
    Who wants to be a millionaire?
  </h2>
</DocumentFragment>
`;
